compiler/rustc_codegen_llvm/src/lib.rs

```diff
@@ -339,24 +339,27 @@ impl CodegenBackend for LlvmCodegenBackend {
         &self,
         ongoing_codegen: Box<dyn Any>,
         sess: &Session,
-    ) -> Result<(Box<dyn Any>, FxHashMap<WorkProductId, WorkProduct>), ErrorReported> {
+    ) -> Result<(CodegenResults, FxHashMap<WorkProductId, WorkProduct>), ErrorReported> {
         let (codegen_results, work_products) = ongoing_codegen
             .downcast::<rustc_codegen_ssa::back::write::OngoingCodegen<LlvmCodegenBackend>>()
             .expect("Expected LlvmCodegenBackend's OngoingCodegen, found Box<Any>")
             .join(sess);
-        Ok((Box::new(codegen_results), work_products))
+
+        sess.time("llvm_dump_timing_file", || {
+            if sess.opts.debugging_opts.llvm_time_trace {
+                llvm_util::time_trace_profiler_finish("llvm_timings.json");
+            }
+        });
```

Comment:
Does this change come from upstream?

Reply:
Yes. https://github.com/rust-lang/rust/blob/master/compiler/rustc_codegen_llvm/src/lib.rs#L347

I don't know how this got removed from our fork. Maybe some rebase / merge conflict resolution?

Reply:
Yeah, maybe. Thanks for catching it!